### PR TITLE
Some random fixes

### PR DESCRIPTION
**Pull request by JesusValera**
# Changes

- Declare the property type in a few method calls.
- Make the `Exceptions` classes final and strict.
- Update the class `PhelCodeException` to make it `abstract`.
- Replace some `new` instances by `self`.
- Small improvs.

### CODE REVIEW

File: src/php/Lang/Struct.php [hunk 1]
```diff
@@ -19,9 +18,9 @@ abstract class Struct extends Table
 
     protected function validateOffset($offset)
     {
-        if (!in_array($offset, $this->getAllowedKeys())) {
+        if (!in_array($offset, $this->getAllowedKeys(), false)) {
             $keyName = Printer::nonReadable()->print($offset);
-            $structName = get_class($this);
+            $structName = self::class;
```
Comment (jenshaase) [inline src/php/Lang/Struct.php:23]: Shouldn't this be `static` (Late static binding)?

Comment (JesusValera) [reply src/php/Lang/Struct.php:23]: Yes, you are right -> https://stackoverflow.com/a/4719222/4988034

Comment (JesusValera) [reply src/php/Lang/Struct.php]: Done. :)

**Comment by JesusValera** [comment]
> The build (psalm) failed after the last merge. Can you fix it?

Yes, I am investigating.

By any reason, `psalm` doesn't like the `static::class;`.
At the moment, I am going to put back the `get_class($this)`.

**Comment by JesusValera** [comment]
Okk, the problem was not the `static::class` but the PR #74.
Let me fix it: :)

**Comment by jenshaase** [comment]
Thanks. It's merged now.

**Pull request merged**